Stevenh/setupshifts

stevenhuang010 opened the pull request:
Set up the shifts queries to support addShift, updateShift, deleteShift, and getShift. Schema has been updated to reflect the correct implementation of Shift. index.tsx is just testing code and should not be merged in. 

Doesn't work for adding shifts to users or any user functionality

Action Items:
For the update call, update the users in usersAssigned if usersAssigned has been changed.
For the delete call, update the users's shiftsAssigned field, since this shift is being deleted.

gregoriiaaa commented:
The addShift function is trying to do two different things: create the shift, and assign the shift to a user by adding it to the user's list of shifts. The addShift function should just create the shift and send it to the firebase, then we can have another function like assignShiftToUser that adds the shift to the user's list of shifts. Because when you first create a shift, it'll just be the basic info without any user's assigned to it yet.

stevenhuang010 commented:
> The addShift function is trying to do two different things: create the shift, and assign the shift to a user by adding it to the user's list of shifts. The addShift function should just create the shift and send it to the firebase, then we can have another function like assignShiftToUser that adds the shift to the user's list of shifts. Because when you first create a shift, it'll just be the basic info without any user's assigned to it yet.

fixed! i removed the shiftsAssigned parameter as an input to addShift, and when uploading it to firebase, it initialize it to an empty list
